app/models/comment.rb

```diff
@@ -1,3 +1,7 @@
 class Comment < ApplicationRecord
-  belongs_to :user
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
```

Comment:
- [ Required ] I think it would be better to add an active rails callback which ensures that the `comments counter` is updated when a comment is added to a post. This will ensure that any time a comment is added to a post the comments _count is updated, I would like to suggest that you add this line of code below `line 3`

```ruby
  after_save :update_post_comments_counter
  
  def update_post_comments_counter
    post.update(comments_count: post.comments.count)
  end
 
```